pkg/controller/statusmanager/pod_status.go

```diff
@@ -175,54 +177,67 @@ func (status *StatusManager) updateNodeStatus(nodeName string, ready bool, reaso
-			if containerStatus.State.Running == nil {
-				ready = false
-				if containerStatus.State.Waiting != nil {
-					if containerStatus.State.Waiting.Message == "" {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Waiting.Reason)
-					} else {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Waiting.Message)
-					}
-				} else if containerStatus.State.Terminated != nil {
-					if containerStatus.State.Terminated.Message == "" {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Terminated.Reason)
-					} else {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Terminated.Message)
-					}
+	containerStatus := corev1.ContainerStatus{}
+	ready := true
+	// messages is used to store the error message from conatiner status,
+	// tmpMsgs is for the status that pod just restarted in a short time
+	var messages, tmpMsgs string
+	nodeName := pod.Spec.NodeName
+	var startedAt *time.Time
+	var lastStartedAt *time.Time
+	// when pod is pending, its ContainerStatuses is empty
+	if len(pod.Status.ContainerStatuses) == 0 {
+		ready = false
+		messages = fmt.Sprintf("nsx-node-agent not running: %s.", pod.Status.Phase)
+	}
+	for _, containerStatus = range pod.Status.ContainerStatuses {
+		if containerStatus.State.Running == nil {
+			ready = false
+			if containerStatus.State.Waiting != nil {
+				if containerStatus.State.Waiting.Message == "" {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Waiting.Reason)
+				} else {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Waiting.Message)
 				}
-
+			} else if containerStatus.State.Terminated != nil {
+				if containerStatus.State.Terminated.Message == "" {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Terminated.Reason)
+				} else {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Terminated.Message)
+				}
+			}
+		} else {
+			// pod is running, but we should check if it just restarted
+			startedAt = &containerStatus.State.Running.StartedAt.Time
+			// there're 3 containers in nsx-node-agent pod, we should check the latest started time
+			if lastStartedAt == nil || (*startedAt).Sub(*lastStartedAt) > 0 {
+				lastStartedAt = startedAt
+			}
+			runningSeconds := int64(now.Sub(*startedAt) / 1000000000)
+			if runningSeconds < operatortypes.TimeBeforeRecoverNetwork {
+				log.Info(fmt.Sprintf("NSX node agent for node %s started before %d seconds", nodeName, runningSeconds))
```

Comment:
would it be useful to specify the container name here too as you do in the line below?

Reply:
Done, thanks